### PR TITLE
Fix some corechecker warnings

### PR DESCRIPTION
**Pull request by MikeGitb**
This PR fixes some of the warnings that the c++ core checker throws in VS2017. Mostly const correctnes "issues" and a few suppressions around pointer arithmetic.

### CODE REVIEW

**Comment by MikeGitb** [comment]
Forgot that attribute syntax is not supported by all supported compilers

File: include/gsl/multi_span [hunk 1]
```diff
@@ -1494,7 +1494,9 @@ public:
 
     constexpr reference operator[](const index_type& idx) const GSL_NOEXCEPT
     {
-        return data_[bounds_.linearize(idx)];
```
Comment (neilmacintosh) [inline include/gsl/multi_span]: From memory, `linearize()` already checks the condition that `idx` is within bounds. That is the point of the `bounds_` object. Why would we double-check the calculation here?

Comment (MikeGitb) [inline include/gsl/multi_span]: @neilmacintosh; No idea. Apparently I wasn't fully awake when makeing this change ;) Sorry for that. I removed the last two commits as they where useless without compiler support for attribute syntax anyway.

**Comment by neilmacintosh** [comment]
Thanks @MikeGitb this is helpful cleanup! As a side note, now we've dropped support for VS 2013, you should be free to add `[[suppress(...)]]` annotations in the future if needed. :-)

**Comment by MikeGitb** [comment]
Yes, that was a bit of unfortunate timing on my part. If no one takes care of it first I might provide another PR when I've the time

**Pull request merged**